pkg/user/parallel_tx_submission.go

```diff
@@ -0,0 +1,497 @@
+	accNum, seqNum, err := QueryAccount(context.Background(), client.conns[0], client.registry, workerAddress)
+	if err != nil {
+		return fmt.Errorf("failed to query worker account %s on chain: %w", worker.accountName, err)
+	}
+
+	// Add account to signer
+	account := NewAccount(worker.accountName, accNum, seqNum)
+	if err := client.signer.AddAccount(account); err != nil {
+		return fmt.Errorf("failed to add worker account %s to signer: %w", worker.accountName, err)
+	}
+
+	// Update worker address
+	worker.address = workerAddress.String()
+
+	return nil
+}
+
+// createWorkerAccount creates a new account in the keyring
+func (client *TxClient) createWorkerAccount(accountName string) error {
+	// Check if account already exists in keyring
+	if _, err := client.signer.keys.Key(accountName); err == nil {
+		return nil // Account already exists
+	}
+
+	// Create new account
+	path := hd.CreateHDPath(sdktypes.CoinType, 0, 0).String()
+	_, _, err := client.signer.keys.NewMnemonic(accountName, keyring.English, path, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	if err != nil {
+		return fmt.Errorf("failed to create account %s in keyring: %w", accountName, err)
+	}
+
+	return nil
+}
+
+// accountNeedsFunding checks if an account needs funding by querying its balance
+func (client *TxClient) accountNeedsFunding(ctx context.Context, address sdktypes.AccAddress) (bool, error) {
+	// Query account balance
+	balance, err := QueryAccountBalance(ctx, client.conns[0], client.registry, address, appconsts.BondDenom)
+	if err != nil {
+		// If account doesn't exist, it needs funding
+		return true, nil
+	}
+
+	// Check if balance is less than the default worker balance
+	// Note: we check for >= DefaultWorkerBalance to avoid re-funding accounts that already have sufficient balance
+	return balance.Amount.Int64() < DefaultWorkerBalance, nil
+}
+
+// hasFeeGrant checks if a fee grant exists between granter and grantee
+func (client *TxClient) hasFeeGrant(ctx context.Context, granter, grantee sdktypes.AccAddress) (bool, error) {
+	feegrantQuery := feegrant.NewQueryClient(client.conns[0])
+	_, err := feegrantQuery.Allowance(ctx, &feegrant.QueryAllowanceRequest{
+		Granter: granter.String(),
+		Grantee: grantee.String(),
+	})
+	if err != nil {
+		// If error contains "not found" or similar, grant doesn't exist
+		return false, nil
+	}
+	return true, nil
+}
+
+// fundAndGrantWorkerAccounts sends funds to worker accounts and sets up fee grants
+func (client *TxClient) fundAndGrantWorkerAccounts(ctx context.Context, workers []*txWorker) error {
+	if len(workers) == 0 {
+		return nil
+	}
+
+	msgs := make([]sdktypes.Msg, 0, len(workers)*2) // Each worker needs up to 2 msgs: send + feegrant
+	totalGasLimit := uint64(0)
+
+	masterAddress := client.defaultAddress
+
+	// Create send messages for funding accounts that need funding
+	for _, worker := range workers {
+		// Get worker address
+		record, err := client.signer.keys.Key(worker.accountName)
```

Comment:
client.Signer.Account() could probably work here since the worker seems to be loaded into signer